app/src/main/java/com/example/filemanager/AllFilesViewModel.kt

```diff
@@ -1,21 +1,33 @@
 package com.example.filemanager
 
 import android.os.Environment
-import android.util.Log
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import java.io.File
 
 class AllFilesViewModel() : ViewModel() {
 
     val files = MutableLiveData<List<ListModel>>()
+    private val rootDirectory: File = Environment.getExternalStorageDirectory()
+    var currentFile: ListModel? = ListModel(rootDirectory)
 
     fun fetchFiles() {
-        val path = Environment.getExternalStorageDirectory().absolutePath ?: return
-        Log.d("PATH", path)
-        val files2 = File(path).listFiles()?.map {
+        files.postValue(listFiles(ListModel(rootDirectory)))
+    }
+
+    fun listFiles(currentItem: ListModel): List<ListModel> {
+        return currentItem.file.listFiles()?.map {
             ListModel(it)
         } ?: emptyList()
-        files.postValue(files2)
+    }
+
+    fun listParentDirectory(): List<ListModel> {
+        val parentFile = currentFile?.file?.parentFile ?: return emptyList()
+        currentFile = ListModel(parentFile)
+        return listFiles(ListModel(parentFile))
+    }
+
+    fun compareDirectories(): Boolean {
+        return currentFile?.file != rootDirectory
```

Comment:
Can create a method in `ListModel` and call it directly on the listModel instance

```
fun isRootDirectory(): Boolean {
    return file == Environment.getExternalStorageDirectory()
}
```
then use it:

`currentFile.isRootDirectory()`